presto-main/src/main/java/io/prestosql/server/security/CertificateAuthenticator.java

```diff
@@ -30,32 +31,45 @@
 {
     private static final String X509_ATTRIBUTE = "javax.servlet.request.X509Certificate";
 
+    private final CertificateAuthenticatorManager authenticatorManager;
     private final UserMapping userMapping;
 
     @Inject
-    public CertificateAuthenticator(CertificateConfig config)
+    public CertificateAuthenticator(CertificateAuthenticatorManager authenticatorManager, CertificateConfig config)
     {
         requireNonNull(config, "config is null");
         this.userMapping = createUserMapping(config.getUserMappingPattern(), config.getUserMappingFile());
+        this.authenticatorManager = requireNonNull(authenticatorManager, "authenticatorManager is null");
+        authenticatorManager.setRequired();
     }
 
     @Override
     public Identity authenticate(HttpServletRequest request)
             throws AuthenticationException
     {
-        X509Certificate[] certs = (X509Certificate[]) request.getAttribute(X509_ATTRIBUTE);
-        if ((certs == null) || (certs.length == 0)) {
+        List<X509Certificate> certificates;
```

Comment:
We can combine the declaration with the initializer since it's only initialized in one place